cmeutils/sampling.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from pymbar import timeseries
+
+def equil_sample(
+        data, threshold_fraction=0.0, threshold_neff=1, conservative=True
+    ):
+    """Returns a statistically independent subset of an array of data.
+
+    Parameters
+    ----------
+    data : numpy.typing.Arraylike
+        1-D time dependent data to check for equilibration.
+    threshold_fraction : float, optional, default=0.8
+        Fraction of data expected to be equilibrated.
+    threshold_neff : int, optional, default=100
+        Minimum amount of effectively correlated samples to consider a_t
+        'equilibrated'.
+    conservative : bool, default=True
+        if set to True, uniformly-spaced indices are chosen with interval
+        ceil(g), where g is the statistical inefficiency.  
+        Otherwise, indices are chosen non-uniformly with interval of
+        approximately g in order to end up with approximately T/g total indices
+
+    Returns
+    -------
+    Tuple containing
```

Comment:
```suggestion
    (numpy.ndarray, numpy.ndarray, int, int)
        Uncorrelated sample, indices of the uncorrelated values from the input array, index of the production region start from the input array, and the number of effective samples
```

Reply:
I'm not sure what is the best way to document the return type. I've also seen
```
tuple (list, int)
```
so if that seems better, feel free to change it up :)